opensearch/catalog-opensearch-endpoint/src/main/java/org/codice/ddf/endpoints/ResourceUriTransformerCatalogFrameworkProxy.java

```diff
@@ -0,0 +1,199 @@
+        ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getEnterpriseResource(request);
+    }
+
+    public Map<String, Set<String>> getEnterpriseResourceOptions(String metacardId)
+        throws ResourceNotFoundException {
+        return catalog.getEnterpriseResourceOptions(metacardId);
+    }
+
+    public ResourceResponse getLocalResource(ResourceRequest request) throws IOException,
+        ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getLocalResource(request);
+    }
+
+    public Map<String, Set<String>> getLocalResourceOptions(String metacardId)
+        throws ResourceNotFoundException {
+        return catalog.getLocalResourceOptions(metacardId);
+    }
+
+    public ResourceResponse getResource(ResourceRequest request, String resourceSiteName)
+        throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getResource(request, resourceSiteName);
+    }
+
+    public Map<String, Set<String>> getResourceOptions(String metacardId, String sourceId)
+        throws ResourceNotFoundException {
+        return catalog.getResourceOptions(metacardId, sourceId);
+    }
+
+    public Set<String> getSourceIds() {
+        return catalog.getSourceIds();
+    }
+
+    public SourceInfoResponse getSourceInfo(SourceInfoRequest sourceInfoRequest)
+        throws SourceUnavailableException {
+        return catalog.getSourceInfo(sourceInfoRequest);
+    }
+
+    public QueryResponse query(QueryRequest query) throws UnsupportedQueryException,
+        SourceUnavailableException, FederationException {
+        return catalog.query(query);
+    }
+
+    public QueryResponse query(QueryRequest queryRequest, FederationStrategy strategy)
+        throws SourceUnavailableException, UnsupportedQueryException, FederationException {
+        return catalog.query(queryRequest, strategy);
+    }
+
+    public BinaryContent transform(Metacard metacard, String transformerId,
+            Map<String, Serializable> requestProperties) throws CatalogTransformerException {
+        return catalog.transform(metacard, transformerId, requestProperties);
+    }
+
+    /**
+     * Converts the resource URIs found in the {@link SourceResponse} {@link Metacard}s using the
+     * {@link ActionProvider} injected in the constructor before delegating the call to the proxied
+     * {@link CatalogFramework} object.
+     */
+    public BinaryContent transform(SourceResponse response, String transformerId,
+            Map<String, Serializable> requestProperties) throws CatalogTransformerException {
+        transform(response);
+        return catalog.transform(response, transformerId, requestProperties);
+    }
+
+    public UpdateResponse update(UpdateRequest updateRequest) throws IngestException,
+        SourceUnavailableException {
+        return catalog.update(updateRequest);
+    }
+
+    private void transform(SourceResponse queryResponse) {
```

Comment:
Hey @lessarderic , I like the idea but I'm a little uneasy about having the proxy. Couple quick things that make me uneasy about this approach --  1) the amount of code that just calls the internal framework and 2) the fact that doing this takes away the endpoint using the osgi framework directly for the framework. This means that different endpoints will be using different implementations of frameworks and I think that violates some of the concepts of having the central framework.

I do like the add-on feature though and think it's definitely useful. Is this something you can see every endpoint using? I know you mentioned CSW, if so it might be worthwhile to look at creating a new type of plugin (pretransformplugin?) and have the catalogframework just call that plugin before doing a transform. This would be acceptable from my POV as we're doing a minor version change (2.7.0) which allows for added functionality in the api.

Hopefully this helps, let me know if you have any questions!